src/header/mod.rs

```diff
@@ -320,10 +320,11 @@ impl Headers {
     }
 
     /// Removes a header from the map, if one existed.
-    /// Returns true if a header has been removed.
-    pub fn remove<H: Header>(&mut self) -> bool {
+    /// Returns the header, if one has been removed.
```

Comment:
This is actually incorrect. `None` may also be returned if the removed header was not parseable.

I wonder if we should return `Option<Raw, Option<H>>` or something like that instead, to enable as many usecases as possible.